core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2676,7 +2676,7 @@ private void analyzeWindowFunctionsAndMeasures(QuerySpecification node, List<Exp
                 String name = windowFunction.getName().toString().toLowerCase(ENGLISH);
                 if (name.equals("lag") || name.equals("lead")) {
                     if (window.getOrderBy().isEmpty()) {
-                        throw semanticException(MISSING_ORDER_BY, (Node) windowFunction.getWindow().get(), "%s function requires an ORDER BY window clause", windowFunction.getName());
+                        throw semanticException(MISSING_ORDER_BY, (Node) windowFunction.getWindow().orElseThrow(), "%s function requires an ORDER BY window clause", windowFunction.getName());
```

Comment:
It's ok. Such an error would be a programming bug, not a response to a user error -- the message and context doesn't need to be consumable by the end user.

Regarding the use of `windowFunction` instead of `windowFunction.getWindow()`, that would result in the error location pointing at the function call instead of the OVER clause, which is closer to where the error actually occurs.